### DISCUSSION TITLE
How to created users with default roles ?

### DISCUSSION THREAD

**Discussion by dvrg** [pallets-eco/flask-security#517] [Q&A] [answered]
Hi, i want to create users from forms register with specified default roles, how I can do it ?

I have try like this but not works

```python
class User(SearchableMixin, db.Model, fsqla.FsUserMixin):
    pass
    first_name = db.Column(db.String(16), nullable=True)
    last_name = db.Column(db.String(16), nullable=True)

    def __init__(self, **kwargs):
        """add default role to user"""
        super(User, self).__init__(**kwargs)
        if self.roles == [] or self.roles is None:
            self.roles = ["SuperUser"]
```

Thanks in advance!

**Comment by jwag956** [comment, marked as answer]
Attach to the user_registered signal, and add the role there:

```
@user_registered.connect_via(app)
def user_registered_sighandler(sender, **extra):
    # Add additional fields to new user - basic role
    logging.getLogger(__name__).debug("register handler: %s", extra)
    user = extra.get("user")
    sender.db.add_role_to_user(user, "basic")
    sender.db.commit()
```

**Reply by dvrg** [reply]
I have tried to implement this code, but show me errors `AttributeError: 'Flask' object has no attribute 'db'` on `sender.db.add_role_to_user(user, "SuperUser")` ? Any suggestions, what's wrong here ? @jwag956 

Thanks in advance!

**Reply by jwag956** [reply]
Sorry - cut and pasted that from my app which does things a bit differently. You need your xxUserDataStore object - depending on which backing DB you use - for example if you use Flask-SQLAlchemy somewhere you did something like:

`mydb = SQLAlchemyUserDatastore(db, User, Role)`

And then passed 'mydb'  that to the Security constructor. Often in your app initialization where you are doing all this - you can put in the signal definition so you can get hold of that object. You can also set an attribute on your application that is that db object (which is what I did and where the example above came from) - then the above code should work.
